index.js

```diff
@@ -527,13 +527,14 @@ const plugin = fp(async function (app, opts) {
     }
 
     // Trigger preExecution hook
+    let modifiedSchema
     let modifiedDocument
     if (context.preExecution !== null) {
-      ({ modifiedDocument } = await preExecutionHandler({ schema: fastifyGraphQl.schema, document, context }))
+      ({ modifiedSchema, modifiedDocument } = await preExecutionHandler({ schema: fastifyGraphQl.schema, document, context }))
     }
 
     // minJit is 0 by default
-    if (shouldCompileJit) {
+    if (shouldCompileJit && !modifiedSchema) {
```

Comment:
This should be documented. I think the same problem might happen for modifiedDocument too - we might end up fixing a bug.

Reply:
Fixed

> I think the same problem might happen for modifiedDocument

I think the `modifiedDocument` is worth being jit'ed it is a consistent update.
For example, a developer is renaming a field from `old` to `new`